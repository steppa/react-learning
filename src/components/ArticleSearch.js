import React, {Fragment} from 'react';
import Header from './Header';
import ArticleList from './ArticleList';
import Input from "./UI/Input";
import Spinner from "./UI/Spinner/Spinner";

class ArticleSearch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchText: '',
      error: null,
      isLoaded: false,
      articles: [],
      isLoading: false
    };
    this.handleSearchTextChange = this.handleSearchTextChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSearchTextChange(e, props) {

    this.setState({searchText: e.target.value});

  }

  handleSubmit(e){
    //this.setState({searchText: this.props.searchText});
    e.preventDefault();
    this.callApi();
  }

  componentDidMount(){
    // this.callApi()
  }



  callApi() {
    this.setState({isLoading: true});
    //TODO: @shousden Change to Axios as that appears to be popular and more simple.
    fetch("https://content.guardianapis.com/search?q="+this.state.searchText+"&from-date=2019-01-01&api-key=bf18382d-5a0b-44cc-9713-4dd5a9d082b4")
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            articles: result.response.results
          });

        },
        // Handle errors don't catch!
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )

  }

  render()
  {
    return (

      <div id='guardianArticleList'>
        <h2>Search the Guardian</h2>
        <p>Use this form to search all Guardian and Observer articles produced in 2019</p>
        <form onSubmit={this.handleSubmit}>
          <Fragment id='searchbar'>
            <Input type={'text'}
                   title= {'Search for Articles: '}
                   name= {'search'}
                   value={this.state.searchText}
                   placeholder = {'Enter a search'}
                   handleChange = {this.handleSearchTextChange}
            />
            <Input type={'submit'}
                   value="Submit"
            />
          </Fragment>
        </form>
        <ArticleList articles={this.state.articles}/>
      </div>

    );
  }

}

export default ArticleSearch;


