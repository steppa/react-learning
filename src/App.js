import React, {Fragment} from 'react';
import './App.css';
import ArticleSearch from './components/ArticleSearch';
import Footer from './components/Footer';
import Home from "./components/Home";
import Header from "./components/Header"
import NavBar from  "./components/NavBar";
import {Route, HashRouter } from "react-router-dom";
import PythonContent from "./components/PythonContent";

class Main extends React.Component {
  render() {
    return(
        <Fragment>
          <Header/>
          <HashRouter>
            <NavBar/>
            <section className="container-fluid" id="contentContainer">

              <div className="col"></div>
              <div className="col">
              <Route exact path="/" component={Home}/>

              <Route path="/GuardianSearch" component={ArticleSearch}/>
              <Route path="/PythonContent" component={PythonContent}/>
              </div>
              <div className="col"></div>
            </section>
          </HashRouter>

          <Footer/>
        </Fragment>
    )
  }
}

export default Main;