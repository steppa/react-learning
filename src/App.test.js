import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import ArticleSearch from './components/ArticleSearch';


it('renders correctly', () => {
  const tree = renderer
    .create(<ArticleSearch/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

