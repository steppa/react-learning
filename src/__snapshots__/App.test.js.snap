// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar it matches snapshot 1`] = `
<div
  id="guardianArticleList"
>
  <h2>
    Search the Guardian
  </h2>
  <p>
    Use this form to search all Guardian and Observer articles produced in 2019
  </p>
  <form
    onSubmit={[Function]}
  >
    <div
      id="searchbar"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="search"
        >
          Search for Articles: 
        </label>
        <input
          className="form-input"
          id="search"
          name="search"
          onChange={[Function]}
          placeholder="Enter a search"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        />
        <input
          className="form-input"
          type="submit"
          value="Submit"
        />
      </div>
    </div>
  </form>
  <article>
    <ul />
  </article>
</div>
`;

exports[`renders correctly 1`] = `
<div
  id="guardianArticleList"
>
  <h2>
    Search the Guardian
  </h2>
  <p>
    Use this form to search all Guardian and Observer articles produced in 2019
  </p>
  <form
    onSubmit={[Function]}
  >
    <div
      id="searchbar"
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="search"
        >
          Search for Articles: 
        </label>
        <input
          className="form-input"
          id="search"
          name="search"
          onChange={[Function]}
          placeholder="Enter a search"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        />
        <input
          className="form-input"
          type="submit"
          value="Submit"
        />
      </div>
    </div>
  </form>
  <article>
    <ul />
  </article>
</div>
`;
